client/packages/system/src/Patient/PatientView/PatientView.tsx

```diff
@@ -230,6 +244,23 @@ const PatientDetailView = ({
     onEdit(isDirty);
   }, [isDirty, onEdit]);
 
+  useEffect(() => {
+    (async () => {
+      if (fromPrescription === true && currentPatient != null) {
+        const invoiceNumber = await createPrescription({
+          id: FnUtils.generateUUID(),
+          patientId: currentPatient.id,
+        });
+        navigate(
+          RouteBuilder.create(AppRoute.Dispensary)
+            .addPart(AppRoute.Prescription)
+            .addPart(String(invoiceNumber))
+            .build()
+        );
+      }
+    })();
+  }, [currentPatient]);
```

Comment:
Do we need another `useEffect`? Can we just add another check and navigate at the end of the `save` method?